File: python/funciones.py
def suma (a, b):
    return a + b

def resta (a, b):
    return a - b

def multiplicacion (a, b):
    return a * b

def divicion (a, b):
    return a / b

def potencia (a, b):
    return a ** b

def calculador():
    while True:
        print("\n---Calculadora en python---")
        print("1. suma")
        print("2. resta")
        print("3. multiplicacion")
        print("4. divicion")
        print("5. potenciacion")
        print("6. Salir")
        
        opcion = input("Selecione una opcion (1-6): ")
        
        if opcion == "6":
            print("Saliendo de la calculadora...")
            break
        
        if opcion not in ["1","2","3","4","5","6"]:
            print("Opcion no valida. Intente de nueevo. ")
            continue
        
        try:
            num1 = float(input("Ingrese el primer numero: "))
            num2 = float(input("Ingrese el segundo numero: "))
            
            operaciones = {
                "1" : suma,
                "2" :resta,
                "3" : multiplicacion,
                "4" : divicion,
                "5" : potencia
            }
            
            resutado = operaciones[opcion](num1, num2)
            print (f"resultado: {resutado}")
            
        except ValueError:
                print("Error: Ingrese solo numerosvalidos. ")
calculador()
    

File: javaScript/funciones.js
// Funciones para las operaciones
function suma(a, b) {
    return a + b;
}

function resta(a, b) {
    return a - b;
}

function multiplicacion(a, b) {
    return a * b;
}

function division(a, b) {
    return a / b;
}

function potencia(a, b) {
    return Math.pow(a, b);
}

// Función de la calculadora
function calculador() {
    while (true) {
        console.log("\n--- Calculadora en JavaScript ---");
        console.log("1. Suma");
        console.log("2. Resta");
        console.log("3. Multiplicación");
        console.log("4. División");
        console.log("5. Potenciación");
        console.log("6. Salir");

        const opcion = prompt("Seleccione una opción (1-6):");

        if (opcion === "6") {
            console.log("Saliendo de la calculadora...");
            break;
        }

        if (!["1", "2", "3", "4", "5", "6"].includes(opcion)) {
            console.log("Opción no válida. Intente de nuevo.");
            continue;
        }

        try {
            const num1 = parseFloat(prompt("Ingrese el primer número:"));
            const num2 = parseFloat(prompt("Ingrese el segundo número:"));

            const operaciones = {
                "1": suma,
                "2": resta,
                "3": multiplicacion,
                "4": division,
                "5": potencia
            };

            const resultado = operaciones[opcion](num1, num2);
            console.log(`Resultado: ${resultado}`);

        } catch (error) {
            console.log("Error: Ingrese solo números válidos.");
        }
    }
}

// Llamar a la función de la calculadora
calculador();